### PR TITLE
Allow use_edns configuration for AsyncClient

### PR DESCRIPTION
**Pull request by astro**
Proposal of a solution to #1491 

Is this the proper place for the option? Please review.

### CODE REVIEW

**Comment by bluejekyll** [comment]
I took a quick look. At first glance this looks like a good change. Let me review tomorrow in more depth.

**Comment by djc** [comment]
Do we really need a toggle on the outside? As in, are there concrete scenarios in which you want to have the same client issue both EDNS and non-EDNS messages?

**Comment by astro** [comment]
What do you mean by *outside*? I use the client crate and need to turn off EDNS so that BIND9 accepts my TSIG-signed update messages as explained in #1491.

File: crates/proto/src/xfer/dns_handle.rs [hunk 1]
```diff
@@ -41,6 +41,11 @@ pub trait DnsHandle: 'static + Clone + Send + Sync + Unpin {
         false
     }
 
+    /// Allow for disabling EDNS
+    fn is_using_edns(&self) -> bool {
```
Comment (bluejekyll) [inline crates/proto/src/xfer/dns_handle.rs:45]: If I'm not mistaken, this is only used in `ClientHandle` contexts, can we instead define it on that?

**Review by bluejekyll — COMMENTED** [review]
I think I only have the one question, otherwise this appears to be a good change.

File: crates/proto/src/xfer/dns_handle.rs (hunk 1 above)
Comment (astro) [reply crates/proto/src/xfer/dns_handle.rs:45]: How? `ClientHandle` has a blanket implementation:
```rust
impl<T> ClientHandle for T where T: DnsHandle<Error = ProtoError> {}
```

Comment (bluejekyll) [reply crates/proto/src/xfer/dns_handle.rs:45]: 3 lines down is the trait implementation... I was suggesting moving the `is_using_edns` from `DnsHandle` to `ClientHandle`. If it's not too hard. That way we can keep this isolated in-case we want to change the way edns is treated in the future.

Comment (astro) [reply crates/proto/src/xfer/dns_handle.rs:45]: The `impl<T> ClientHandle` can only access what `trait DnsHandle` provides, so ` is_using_edns()` would have to be defined in `DnsHandle` anyway.

Comment (bluejekyll) [reply crates/proto/src/xfer/dns_handle.rs:45]: yes, you're correct.

**Review by bluejekyll — APPROVED** [review]

**Comment by bluejekyll** [comment]
Thanks for the PR! I'll take care of the fmt issue.

**Pull request merged**